components/Audio/index.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react'
+import { AudioPlayer } from '@project-r/styleguide'
+import withT from '../../lib/withT'
+import { compose } from 'react-apollo'
+import { css } from 'glamor'
+import { mediaQueries, zIndex } from '@project-r/styleguide'
+import ProgressComponent from '../../components/Article/Progress'
+
+export const AudioContext = React.createContext({
+  audioState: {},
+  toggleAudioPlayer: () => {}
+})
+
+export const AudioProvider = ({ children, t }) => {
+  const [audioState, setAudioState] = useState(undefined)
+  const [audioPlayerVisible, setAudioPlayerVisible] = useState(false)
+
+  const toggleAudioPlayer = payload => {
+    setAudioPlayerVisible(true)
+    setAudioState(payload)
+  }
+
+  const onCloseAudioPlayer = () => {
+    setAudioPlayerVisible(false)
+    setTimeout(() => {
+      setAudioState(undefined)
+    }, 300)
+  }
```

Comment:
Move the `{audioState && (` up around the first `div`. And then do the following:

```suggestion
  const [audioState, setAudioState] = useState(undefined)
  const [audioPlayerVisible, setAudioPlayerVisible] = useState(false)

  const toggleAudioPlayer = payload => {
    setAudioState(payload, () => {
      setAudioPlayerVisible(true)
    })
  }
  const onCloseAudioPlayer = () => {
    setAudioPlayerVisible(false)
  }
  
  useEffect(() => {
    if (!audioPlayerVisible && audioState) {
      const timeout = setTimeout(() => {
        setAudioState(undefined)
      }, 300)
      return () => {
        clearTimeout(timeout)
      }
    }
  }, [audioState, audioPlayerVisible])
```

The `useEffect` ensures that if the `audioState` changes in the 300ms fade out that `setAudioState(undefined)` is no longer called.